GitHubActions/context.ps1

```diff
@@ -0,0 +1,91 @@
+
+## This commands implement a GitHub Actions 'Context' enacapsulating
+## various elements of the running environment of the current action.
+## It is an adaptation of the TypeScript version found here:
+##  https://github.com/actions/toolkit/blob/master/packages/github/src/context.ts
+
+
+## For now, every access to the Action Context requires constructing a
+## new one since it's implemented purely as a nested tree of hashtables
+## so there is no way to prevent modification -- future enhancement may
+## be to re-implement as custom type with more strict, read-only members
+## so we can resolve a globally shared instance
+function New-ActionContextMap {
```

Comment:
Ideally the Context should be cached so it's not calculated over and over... or at least the Payload part.

Then `Get-ActionRepo` and `Get-ActionIssue` don't have to create additional contexts

Reply:
Is `Map` at the end really needed? I feel like it doesn't provide much value.

Reply:
> Ideally the Context should be cached so it's not calculated over and over... or at least the Payload part.
> 
> Then `Get-ActionRepo` and `Get-ActionIssue` don't have to create additional contexts

Yes, I agree, but this speaks to my first bullet comment above -- that the context should be immutable to start with so that it can be re-used multiple times, otherwise, any modifications will impact subsequent references.

Reply:
> Is `Map` at the end really needed? I feel like it doesn't provide much value... instead you could specify an `OutputType`

So this my attempt to flag or call out that it's a prototype based on hashtables, versus the ultimate approach that may be based on classes or `pscustomobject`.  It's just temporary.